### PR TITLE
provider/aws: Add support for account password policy

### PR DESCRIPTION
**Pull request by radeksimko**
This should allow codifying AWS account-wide password policy.

To do:

 - [x] Initial implementation
 - [x] Documentation
 - [x] Acceptance tests

### CODE REVIEW

File: builtin/providers/aws/resource_aws_iam_account_password_policy.go [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+	d.Set("password_reuse_prevention", policy.PasswordReusePrevention)
+	d.Set("require_lowercase_characters", policy.RequireLowercaseCharacters)
+	d.Set("require_numbers", policy.RequireNumbers)
+	d.Set("require_symbols", policy.RequireSymbols)
+	d.Set("require_uppercase_characters", policy.RequireUppercaseCharacters)
+
+	return nil
+}
+
+func resourceAwsIamAccountPasswordPolicyUpdate(d *schema.ResourceData, meta interface{}) error {
```
Comment (stack72) [inline builtin/providers/aws/resource_aws_iam_account_password_policy.go]: I don't think we need to do this. In the schema declaration, we can do this:

```
Create: resourceAwsIamAccountPasswordPolicyCreate,
Read:   resourceAwsIamAccountPasswordPolicyRead,
Update: resourceAwsIamAccountPasswordPolicyCreate,
Delete: resourceAwsIamAccountPasswordPolicyDelete,
```

**Comment by stack72** [comment]
This one is also ace! If you want, you can merge and I can add acceptance tests as I have some free time today

**Comment by radeksimko** [comment]
This is now ready for review.

**Comment by stack72** [comment]
@radeksimko this looks really good. The only small thing I would say is that we should maybe add a note to the docs saying that only 1 password policy is allowed per AWS account. Apart from that it's :+1: 

```
make testacc TEST=./builtin/providers/aws TESTARGS='-run=TestAccAWSIAMAccountPasswordPolicy'
==> Checking that code complies with gofmt requirements...
go generate $(go list ./... | grep -v /vendor/)
TF_ACC=1 go test ./builtin/providers/aws -v -run=TestAccAWSIAMAccountPasswordPolicy -timeout 120m
=== RUN   TestAccAWSIAMAccountPasswordPolicy_basic
--- PASS: TestAccAWSIAMAccountPasswordPolicy_basic (11.90s)
PASS
ok  	github.com/hashicorp/terraform/builtin/providers/aws	11.918s
```

**Comment by radeksimko** [comment]
Agreed about the note, I added the following:

screen shot 2016-02-26 at 13 30 26

I will wait for green from Travis, then merge.

**Comment by stack72** [comment]
Spot on :)

**Pull request merged**